ports/stm32/mpconfigboard_common.h

```diff
@@ -52,6 +52,16 @@
 #define MICROPY_PY_PYB_LEGACY (1)
 #endif
 
+// Whether to include legacy methods and constants in machine.Pin (which is also pyb.Pin).
+#ifndef MICROPY_PY_MACHINE_PIN_LEGACY
+#define MICROPY_PY_MACHINE_PIN_LEGACY (1)
```

Comment:
Do we want to phase out these legacy items? If so, could make the default here `!MICROPY_PREVIEW_VERSION_2`

Reply:
Yes, good idea, updated as suggested.